### DISCUSSION TITLE
Looking at Vite.js?

### DISCUSSION THREAD

**Discussion by richb-hanover** [PRQL/prql#3459] [General]
Is anyone looking at [Vite.js](https://vitejs.dev/)?

Its claim to fame is that it relies on ES Modules to skip the process of building an entire binary Javascript thingie during development. Instead the browser uses hot-reloading to pull in the files that have changed, taking fractions of a second to update when a source file changes. 

My reading of the documentation is that it's straightforward to take a React project and switch the tooling to Vite. And for production builds, Vite uses `esbuild` as an alternative to Webpack. `esbuild` is notable for being an order of magnitude faster building the final distribution files.

I might take a look at this in my spare time, if no one else is interested in tackling the project. Thanks

**Comment by max-sixty** [comment]
Generally, it would be great to replace the current [`wasm-react-scripts`](https://github.com/PRQL/prql/blob/455c54d365bfd502d66e1b07217ae99d3c140fad/web/playground/package.json#L25) with something else — this is forked version of another package which isn't supported and has lots of old dependencies. IIUC it uses webpack.

I think Vite may be interesting — I don't know enough to have a view on whether it's the "best" solution, but likely it's at least a reasonable choice. [ruff's playground](https://github.com/astral-sh/ruff/tree/main/playground) uses it.

**Comment by vanillajonathan** [comment]
Yes, Vite is quite nice and it is easy to switch to the tooling.

Something else that is nice is [Bun](https://bun.sh/) which is a drop-in replacement for Node.js, so it is very easy to use too, and it is very fast.

**Comment by richb-hanover** [comment]
@vanillajonathan I had delusions of having time to think about this, but "stuff" keeps cropping up. If either of these is interesting, feel free to jump in. Otherwise, we can leave this idea in the queue until [we get around to it.](https://www.thecrackedpig.com/product/round-tuit-wooden-token/) :-)

**Comment by max-sixty** [comment]
I'd be fine with bun / vite if they work well! FWIW I think most of the build time is taken up by `wasm-opt` rather than the npm build steps, though those are material too.

I'm be predominantly interested in replacing the [wasm-react-scripts](https://github.com/PRQL/prql/blob/455c54d365bfd502d66e1b07217ae99d3c140fad/web/playground/package.json#L25) dependency, since that's the piece that's not maintained. To the extent those are correlated (or someone would just prefer to do the work on the `npm`-equivalent side), then great.

**Comment by richb-hanover** [comment]
For any brave archaeologists out there... Look at https://github.com/Cosmian/create-react-app/commit/cb7608df5618a7bdc7e3484c6192d8249f9bc1ac for an upgrade to wasm-react-scripts 5.0.4

**Reply by max-sixty** [reply]
Ah, good point, I just put in #3565 to upgrade it

**Comment by richb-hanover** [comment]
To close this out, we recently incorporated Vite.js  into the build system. It works great, and seems to have removed the reliance on wasm-react-scripts
